fs/binfs/fs_binfs.c

```diff
@@ -326,7 +326,7 @@ static int binfs_readdir(struct inode *mountpt, struct fs_dirent_s *dir)
 
       finfo("Entry %d: \"%s\"\n", index, name);
       dir->fd_dir.d_type = DTYPE_FILE;
-      strncpy(dir->fd_dir.d_name, name, NAME_MAX);
+      strlcpy(dir->fd_dir.d_name, name, NAME_MAX);
```

Comment:
Should these all be `NAME_MAX + 1` because strlcpy copies at most `dsize-1` and `d_name` is defined as with a size of `NAME_MAX + 1`?

Reply:
Actually there's a better way, use `sizeof()`, that way seems less error prone.

Reply:
Good suggestion, please adjust patch @DS-LK .

Reply:
OK.

Reply:
Done.